.github/workflows/update-priority.yml

```diff
@@ -0,0 +1,63 @@
+# Updates the Priority field on a project board when a [Pri] label gets added to an issue.
+name: Update Priority Field
+
+# Triggers the workflow when a label is added to an issue.
+on:
+  issues:
+    types:
+      - labeled
+
+# Details needed to update a project field.
+env:
+  LABEL_PREFIX: "[Pri] "
+  PROJECT_NUMBER: 391
+  PROJECT_FIELD: "Priority"
+  ORGANIZATION: "Automattic"
+
+jobs:
+
+  # Updates the Priority field with the appropriate priority level.
+  update_priority: 
+
+    runs-on: ubuntu-latest
+
+    # Only runs if a priority label is added.
+    if:  contains(github.event.label.name, '[Pri]')
```

Comment:
This only runs when `[Pri]` labels are added, which cuts down on unnecessary runs. I originally wanted to also skip the entire action if the issue wasn't in the project, but there doesn't seem to be an easy way to get that info from a single issue event without iterating through the entire project board first 😢 It's not a dealbreaker though! It won't cause any chaos, just extra runs. 

Also, I really wish I could use the LABEL_PREFIX environment variable I created above here, but it's just one of those [weird GitHub Actions quirks again](https://github.com/actions/runner/issues/1189#issuecomment-880110759).